docs/patterns/CVE-2018-20992-HIR.rpl

```diff
@@ -0,0 +1,42 @@
+pattern CVE-2018-20992
+
+patt {
+    use std::vec::Vec;
+
+    // The purpose of the middle brackets seems to be
+    // declaring metavariables for types that are not `identifiers`.
+    p_vec_set_len_to_extend[
+        $T: ty
+    ] = 
+        pub fn _ (..) -> _ {
+            @let $vec: Vec<$T> = _;
+            @let $new_len: usize = _;
+            @let $old_len: usize = $vec.len();
+            #without!(if $new_len <= $old_len) {
+                $vec.set_len($new_len);
+            }
+        }
+}
+
+// if we consider the `#without!` like `#or!` as a String Substitution.
```

Comment:
Some thoughts about `#without!` macro, try to consider it as a string substitution operation.
@frank-king @TheVeryDarkness